sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala

```diff
@@ -125,6 +135,31 @@ class CatalogImpl(sparkSession: SparkSession) extends Catalog {
       isTemporary = isTemp)
   }
 
+  private def makeTable(ident: Seq[String]): Table = {
+    val plan = UnresolvedTableOrView(ident, "Catalog.listTables", true)
+    val node = sparkSession.sessionState.executePlan(plan).analyzed
+    node match {
+      case t: ResolvedTable =>
+        val isExternal = t.table.properties().getOrDefault("external", "false").equals("true")
+        new Table(
+          name = t.identifier.name(),
+          database = t.identifier.namespace().head,
+          description = t.table.properties().get("comment"),
+          tableType =
+            if (isExternal) CatalogTableType.EXTERNAL.name
+            else CatalogTableType.MANAGED.name,
+          isTemporary = false)
+      case v: ResolvedView =>
+        new Table(
+          name = v.identifier.name(),
+          database = v.identifier.namespace().toString,
+          description = "",
+          tableType = "",
```

Comment:
looking at the previous behavior, I think this should be `if (v.isTemp) "TEMPORARY" else "VIEW"`

Reply:
Good catch!